### PR TITLE
Set the correct op flags on new OP_ISBOOL

### PR DESCRIPTION
**Pull request by leonerd**

### CODE REVIEW

File: t/perf/opcount.t [hunk 1]
```diff
@@ -696,10 +696,19 @@ test_opcount(0, "builtin::true/false are replaced with constants",
                 });
 
 test_opcount(0, "builtin::isbool is replaced with direct opcode",
-                sub { my $x = 123; my $y = builtin::isbool($x); },
+                sub { my $x = 123; my $y; $y = builtin::isbool($x); },
                 {
                     entersub => 0,
                     isbool   => 1,
+                    padsv    => 3, # OA_TARGLEX applies so only 3, not 4
```
Comment (ilmari) [inline t/perf/opcount.t:703]: Should this also check `sassign => 0`?

Comment (leonerd) [reply t/perf/opcount.t:703]: Well, 1 because `$x` still has one. Just not `$y`. But yes I'll add that. 2df1a36e41

Comment (ilmari) [reply t/perf/opcount.t:703]: You could avoid the `$x` one by leaving it uninitialised, since `isbool(undef)` is perfectly valid.

Comment (leonerd) [reply t/perf/opcount.t:703]: Ohyes, I just wanted something that wouldn't const-fold. That will do too. 5ea6acf8e9

**Review by ilmari — APPROVED** [review]

File: pp.c [hunk 2]
```diff
@@ -7215,9 +7215,11 @@ PP(pp_cmpchain_dup)
 PP(pp_isbool)
 {
     dSP;
+    dTARGET;
     SV *arg = POPs;
 
-    PUSHs(boolSV(SvIsBOOL(arg)));
+    sv_setbool(TARG, SvIsBOOL(arg));
```
Comment (ilmari) [inline pp.c]: Shouldn't this be `sv_setbool_mg`?

**Review by ilmari — CHANGES_REQUESTED** [review]

File: pp.c (hunk 2 above)
Comment (leonerd) [reply pp.c]: It should. And we should SvGETMAGIC() beforehand as well. I'll add those + some tests for it.

Comment (leonerd) [reply pp.c]: Magic added in c2d575565b

**Review by ilmari — APPROVED** [review]

**Pull request merged**